iFixit/react-commerce
# Add bulk FAQs assignment

closes #1940

### Summary

This PR introduces the first iteration of bulk assignment for FAQs. The key features are:

#### FAQ Linking
- An FAQ can be linked to multiple product lists.
  - The FAQ will be visible on every linked product list as well as their descendants, assuming an FAQs section is present. A reusable FAQs section [is already configured in this brannch](https://implement-reusable-faqs.govinor.com/admin/content-manager/collectionType/api::reusable-section.reusable-section/3).

#### FAQ Categories
- FAQs can belong to a category.
  - On each product list, only one FAQ per category is displayed. Descendant FAQs take precedence over ancestor FAQs.
  - All FAQs without a category are displayed.

#### Item Type Filter
- FAQs can be filtered by item type (e.g., "Batteries").
  - If set, the FAQ will be displayed only if the product list contains items of that type (as verified by the presence of the item type in the filter list).

#### Priority
- FAQs can be prioritized.
  - FAQs with higher priority values will appear higher on the list.

### QA Steps

Use both the [Vercel preview](https://react-commerce-git-implement-reusable-faqs-ifixit.vercel.app/Parts) and the [Strapi preview](https://implement-reusable-faqs.govinor.com/admin/content-manager/collectionType/api::faq.faq?page=1&pageSize=10&sort=question:ASC) to verify that the features listed above are correctly implemented.

cc @sterlinghirsh @jeffsnyder
